pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java

```
@@ -391,10 +391,18 @@ public Representation updateTableConfig(@Parameter(name = "tableName", in="path"
 
       _pinotHelixResourceManager.setExistingTableConfig(config, tableNameWithType, tableType);
       return responseRepresentation(Status.SUCCESS_OK, "{\"status\" : \"Success\"}");
+    } catch(PinotHelixResourceManager.InvalidTableConfigException e) {
+      LOGGER.info("Failed to update configuration for table {}, message: {}", tableName, e.getMessage());
+      ControllerRestApplication.getControllerMetrics().addMeteredGlobalValue(
+          ControllerMeter.CONTROLLER_TABLE_UPDATE_ERROR, 1L);
+      return errorResponseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST, e.getMessage());
     } catch (IOException e) {
       LOGGER.error("Failed to update table configuration for table: {}", tableName, e);
+      ControllerRestApplication.getControllerMetrics().addMeteredGlobalValue(
+          ControllerMeter.CONTROLLER_TABLE_UPDATE_ERROR, 1L);
       return errorResponseRepresentation(Status.SERVER_ERROR_INTERNAL,
           "Internal error while updating table configuration");
+
```

Do we want to add a catch all exception here?